src/main/java/sg/ncl/MainController.java

```diff
@@ -2105,113 +2102,123 @@ public String data(Model model, HttpSession session) throws Exception {
-//    		redirectAttributes.addFlashAttribute("editPublic", "success");
-//    	}
-//
-//    	if (origDataset.updateAuthorization(editedDatasetIsRequiredAuthorization) == true) {
-//    		redirectAttributes.addFlashAttribute("editIsRequiredAuthorization", "success");
-//    	}
-//
-//    	datasetManager.updateDatasetDetails(origDataset);
-//
-//    	return "redirect:/data/edit/{datasetId}";
-//    }
+                switch (exceptionState) {
+                    case DATASET_NAME_IN_USE_EXCEPTION:
+                        log.error("Dataset name already exists.");
+                        model.addAttribute("message", "Error(s):<ul><li>dataset name already exists</li></ul>");
+                        break;
+                    case FORBIDDEN_EXCEPTION:
+                        log.error("Saving of dataset forbidden.");
+                        model.addAttribute("message", "Error(s):<ul><li>saving dataset forbidden</li></ul>");
+                        break;
+                }
 
-//    @RequestMapping("/data/remove_dataset/{datasetId}")
-//    public String removeDataset(@PathVariable Integer datasetId) {
-//    	datasetManager.removeDataset(datasetId);
-//    	return "redirect:/data";
-//    }
+                return CONTRIBUTE_DATA_PAGE;
+            }
+        } catch (IOException e) {
+            throw new WebServiceRuntimeException(e.getMessage());
+        }
+
+        return "redirect:/data";
+    }
+
+    private ResponseEntity getResponseEntity(@PathVariable Optional<String> id, HttpEntity<String> request) {
+        ResponseEntity response;
+        if (!id.isPresent()) {
+            response = restTemplate.exchange(properties.getSioDataUrl(), HttpMethod.POST, request, String.class);
+        } else {
+            response = restTemplate.exchange(properties.getDataset(id.get()), HttpMethod.PUT, request, String.class);
+        }
+        return response;
+    }
+
+    @RequestMapping("/data/remove/{id}")
+    public String removeDataset(@PathVariable String id, RedirectAttributes redirectAttributes) throws WebServiceRuntimeException {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getDataset(id), HttpMethod.DELETE, request, String.class);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                if (exceptionState == FORBIDDEN_EXCEPTION) {
+                    log.error("Removing of dataset forbidden.");
+                    redirectAttributes.addFlashAttribute("message", error.getMessage());
+                }
+            }
+        } catch (IOException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1166)